### PR TITLE
relace contextfilter with pass_context

### PR DESCRIPTION
**Pull request by YuxingW**
contextfilter is deprecated in jinja2 and can be replaced by pass_context
https://github.com/jupyter/nbconvert/issues/1568
Can we please replace contextfilter in filter_plugins/filters.py? Otherwise there is an error with new jinja2:
michaelrigart.interfaces/filter_plugins/filters.py) as it seems to be invalid: module 'jinja2' has no attribute 'contextfilter'

### CODE REVIEW

File: filter_plugins/filters.py [hunk 1]
```diff
@@ -128,7 +128,7 @@ def _interface_check(context, interface, interface_type=None):
     return _pass()
 
 
-@jinja2.contextfilter
+@jinja2.pass_context
```
Comment (markgoddard) [inline filter_plugins/filters.py:131]: This breaks on versions of Jinja2 older than 3.0.0. What I've done elsewhere is this:

```
# NOTE: jinja2 3.1.0 dropped contextfilter in favour of pass_context.
try:
    from jinja2 import pass_context
except ImportError:
    from jinja2 import contextfilter as pass_context
```

**Review by markgoddard — CHANGES_REQUESTED** [review]
Thanks

File: filter_plugins/filters.py (hunk 1 above)
Comment (YuxingW) [reply filter_plugins/filters.py:131]: That is something I am also in favor of, thanks!

**Review by markgoddard — APPROVED** [review]
Thanks

**Comment by YuxingW** [comment]
Cool, thanks for the merge.

**Pull request merged**